### PR TITLE
docker-compose development workflow

### PR DESCRIPTION
**Pull request by kitchen**
I don’t use VSCode. Nothing against it, I just use a different environment. And I don’t like globally installing build deps and such, so I much prefer containerized environments! So I’ve tried to reproduce the VSCode devcontainer workflow here as much as possible.

### CODE REVIEW

**Comment by kitchen** [comment]
draft comment: still working on the docs and such.

meta comment: the dev environment docs are visually quite difficult to navigate, imo, and we might want to re-envision them somehow. I have some ideas, but I think they would be out of scope for this PR. They could certainly be a dependency for this PR, however :)

File: docs/docs/development/build-flash.md
Comment (petejohanson) [inline docs/docs/development/build-flash.md:95]: This needs proper capitalization, also for anyone that isn't a developer, or doesn't know what a "containerized environment" is, this might be really confusing.

Perhaps something like:

```suggestion
#### Local/Host Build Environment.
```

File: docs/docs/development/build-flash.md
Comment (petejohanson) [inline docs/docs/development/build-flash.md:125]: Headers should use caps:

```suggestion
#### Docker Compose Build Environment
```

File: docs/docs/development/setup.md
Comment (petejohanson) [inline docs/docs/development/setup.md:19]:
```suggestion
{label: 'Docker Compose', value: 'dockercompose'},
```

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few thoughts on the docs.

**Comment by kitchen** [comment]
ok, this should be ready to roll now, I just tested the `docker compose run -v` on my mac and it seems to be working as expected now!

**Comment by smeikx** [comment]
What is holding this back? Just merged it locally with `main`, read the docs and successfully built the .uf2 files for kyria. 
Works perfectly fine, I am really glad this work has already been done.

File: docs/docs/development/build-flash.md
Comment (caksoylar) [inline docs/docs/development/build-flash.md:97]:
```suggestion
Instead of building .uf2 files using the default keymap and config files, you can build directly from your [`zmk-config` folder](../user-setup.md#github-repo) by adding
```
I think this was changed by a PR by @dxmh, so the diff should be cleaner with the `.md` added back.

**Review by caksoylar — APPROVED** [review]
Since this is confirmed working by an additional user, I think it's good to merge pending this tiny fix.

**Comment by LeducH** [comment]
@kitchen Why are you removing the container after execution `docker compose run --rm .....` ? 

It seems to me it could be good the keep them up since maybe a accidental `crtl + d` throws you out then . 

In any case great job with the suggestions. All the power to docker

File: docker-compose.yml
Comment (chrissicool) [inline docker-compose.yml:2]: My Debian 11.5 complains about the version being too high. 

```
$ docker-compose -v
docker-compose version 1.25.0, build unknown
$ docker --version
Docker version 20.10.5+dfsg1, build 55c4c88
```

Using version: "3" does the trick for me. Works like a charm. And it also has to be 'docker-compose' in the docs.

I assume you are using a newer version of Docker. Does it work for you, if you switch these bits, so that older systems can use this as well?

**Comment by Nick-Munnich** [comment]
Closing as I believe this was made redundant by #2548. Feel free to reopen if you feel otherwise.

**Pull request closed without merging**
